doc/OnlineDocs/contributed_packages/incidence/tutorial.dm.rst

```diff
@@ -0,0 +1,182 @@
+.. _incidence_tutorial_dm:
+
+Debugging a structural singularity with the Dulmage-Mendelsohn partition
+========================================================================
+
+We start with some imports and by creating a Pyomo model we would like
+to debug. Usually the model is much larger and more comlicated than this.
+This particular system appeared when debugging a dynamic 1-D partial
+differential-algebraic equation (PDAE) model representing a chemical looping
+combustion reactor.
+
+.. doctest::
+   :skipif: not scipy_available or not networkx_available or not asl_available
+
+   >>> import pyomo.environ as pyo
+   >>> from pyomo.contrib.incidence_analysis import IncidenceGraphInterface
+
+   >>> m = pyo.ConcreteModel()
+   >>> m.components = pyo.Set(initialize=[1, 2, 3]) 
+   >>> m.x = pyo.Var(m.components, initialize=1.0/3.0)
+   >>> m.flow_comp = pyo.Var(m.components, initialize=10.0)
+   >>> m.flow = pyo.Var(initialize=30.0)
+   >>> m.density = pyo.Var(initialize=1.0)
+   >>> m.sum_eqn = pyo.Constraint(
+   ...     expr=sum(m.x[j] for j in m.components) - 1 == 0
+   ... )
+   >>> m.holdup_eqn = pyo.Constraint(m.components, expr={
+   ...     j: m.x[j]*m.density - 1 == 0 for j in m.components
+   ... })
+   >>> m.density_eqn = pyo.Constraint(
+   ...     expr=1/m.density - sum(1/m.x[j] for j in m.components) == 0
+   ... )
+   >>> m.flow_eqn = pyo.Constraint(m.components, expr={
+   ...     j: m.x[j]*m.flow - m.flow_comp[j] == 0 for j in m.components
+   ... })
+
+To check this model for structural singularity, we apply the Dulmage-Mendelsohn
+partition.
+
+.. doctest::
+   :skipif: not scipy_available or not networkx_available or not asl_available
+
+   >>> igraph = IncidenceGraphInterface(m)
+   >>> # Make sure we have a square system
+   >>> print(len(igraph.variables))
+   8
+   >>> print(len(igraph.constraints))
+   8
+   >>> var_dmp, con_dmp = igraph.dulmage_mendelsohn()
```

Comment:
You use `var_dmp` and `con_dmp` throughout.  What is `dmp` an abbreviation for?  Would it make more sense to spell it out here?  Maybe `var_partition` and `con_partition`?